File: server/models/Milestone.js
const mongoose = require('mongoose');

const milestoneSchema = new mongoose.Schema(
  {
    day: {
      type: Number,
      required: true,
    },
    label: {
      type: String,
      required: true,
      trim: true,
    },
    title: {
      type: String,
      required: true,
      trim: true,
    },
    body: {
      type: String,
      required: true,
      trim: true,
    },
    imageUrl: {
      type: String,
      default: '',
    },
    icon: {
      type: String,
      default: 'local_florist',
    },
    iconFill: {
      type: Boolean,
      default: false,
    },
    colorScheme: {
      type: String,
      enum: ['primary', 'secondary', 'tertiary'],
      default: 'primary',
    },
    aspectRatio: {
      type: String,
      enum: ['video', 'square', '4/5'],
      default: 'video',
    },
  },
  {
    timestamps: true,
  }
);

milestoneSchema.index({ day: 1 });

module.exports = mongoose.model('Milestone', milestoneSchema);

File: App/app/src/main/java/com/ourbloom/app/data/models/Milestone.kt
package com.ourbloom.app.data.models

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Milestone(
    @DocumentId val id: String = "",
    val coupleId: String = "",
    val day: Int = 0,
    val label: String = "",
    val title: String = "",
    val body: String = "",
    val imageUrl: String = "",
    val icon: String = "local_florist",
    val iconFill: Boolean = false,
    val colorScheme: String = "primary",
    val aspectRatio: String = "video"
)
